src/NServiceBus.AcceptanceTests/Sagas/When_saga_started_concurrently.cs

```diff
@@ -0,0 +1,172 @@
+                {
+                    Data.OrderId = message.SomeId;
+                    Data.Placed = true;
+                    Bus.SendLocal(new SuccessfulProcessing
+                    {
+                        SagaId = Data.Id,
+                        Type = nameof(StartMessageOne)
+                    });
+                    CheckForCompletion();
+                }
+
+                public void Handle(StartMessageTwo message)
+                {
+                    Data.OrderId = message.SomeId;
+                    Data.Billed = true;
+                    Bus.SendLocal(new SuccessfulProcessing
+                    {
+                        SagaId = Data.Id,
+                        Type = nameof(StartMessageTwo)
+                    });
+                    CheckForCompletion();
+                }
+
+                protected override void ConfigureHowToFindSaga(SagaPropertyMapper<ConcurrentlyStartedSagaData> mapper)
+                {
+                    mapper.ConfigureMapping<StartMessageOne>(msg => msg.SomeId).ToSaga(saga => saga.OrderId);
+                    mapper.ConfigureMapping<StartMessageTwo>(msg => msg.SomeId).ToSaga(saga => saga.OrderId);
+                }
+
+                void CheckForCompletion()
+                {
+                    if (!Data.Billed || !Data.Placed)
+                    {
+                        return;
+                    }
+                    MarkAsComplete();
+                    Context.SagaCompleted = true;
+                }
+            }
+
+            class ConcurrentlyStartedSagaData : ContainSagaData
+            {
+                [Unique]
+                public virtual string OrderId { get; set; }
+                public virtual bool Placed { get; set; }
+                public virtual bool Billed { get; set; }
+            }
+
+            // Intercepts the messages sent out by the saga
+            class LogSuccessfulHandler : IHandleMessages<SuccessfulProcessing>
+            {
+                public Context Context { get; set; }
+
+                public void Handle(SuccessfulProcessing message)
+                {
+                    if (message.Type == nameof(StartMessageOne))
+                    {
+                        Context.PlacedSagaId = message.SagaId;
+                    }
+                    else if (message.Type == nameof(StartMessageTwo))
+                    {
+                        Context.BilledSagaId = message.SagaId;
+                    }
+                    else
+                    {
+                        throw new Exception("Unknown type");
+                    }
+                }
+            }
+        }
+
+        [Serializable]
+        class StartMessageOne : ICommand
+        {
+            public string SomeId { get; set; }
+        }
+
+        [Serializable]
+        class StartMessageTwo : ICommand
+        {
+            public string SomeId { get; set; }
+        }
+
+        [Serializable]
```

Comment:
Not a 100% https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.AcceptanceTests/Audit/When_a_replymessage_is_audited.cs#L112  and there's probably more. But as I said nitpick, I'll make sure we remove them for v6.1